src/udev/udev-event.c

```diff
@@ -932,33 +932,43 @@ int udev_event_execute_rules(UdevEvent *event,
 
         r = device_clone_with_db(dev, &event->dev_db_clone);
         if (r < 0)
-                log_device_debug_errno(dev, r, "Failed to clone sd_device object, ignoring: %m");
+                return log_device_debug_errno(dev, r, "Failed to clone sd_device object: %m");
 
         if (event->dev_db_clone && sd_device_get_devnum(dev, NULL) >= 0)
                 /* Disable watch during event processing. */
                 (void) udev_watch_end(event->dev_db_clone);
 
-        if (action == DEVICE_ACTION_MOVE)
-                (void) udev_event_on_move(event);
+        if (action == DEVICE_ACTION_MOVE) {
+                r = udev_event_on_move(event);
+                if (r < 0)
+                        return r;
+        }
 
-        (void) udev_rules_apply_to_event(rules, event, timeout_usec, properties_list);
+        r = udev_rules_apply_to_event(rules, event, timeout_usec, properties_list);
+        if (r < 0)
+                return log_device_debug_errno(dev, r, "Failed to apply udev rules: %m");
```

Comment:
Can this ever fail because of bad rules?

Reply:
I hope no. This fails when something internal critical errors like `sd_device_get_xxx()` fails with non-`ENOENT` value.